src/Nethermind/Nethermind.Serialization.Ssz/Ssz.Decode.cs

```diff
@@ -0,0 +1,267 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Buffers.Binary;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Nethermind.Int256;
+
+namespace Nethermind.Serialization.Ssz;
+
+/// <summary>
+/// https://github.com/ethereum/eth2.0-specs/blob/dev/specs/simple-serialize.md#simpleserialize-ssz
+/// </summary>
+public static partial class Ssz
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static void Decode(ReadOnlySpan<byte> span, out bool result)
+    {
+        result = span[0] != 0;
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static void Decode(ReadOnlySpan<byte> span, out byte result)
+    {
+        const int expectedLength = 1;
+        if (span.Length != expectedLength)
+        {
+            throw new InvalidDataException(
+                $"{nameof(DecodeByte)} expects input of length {expectedLength} and received {span.Length}");
+        }
```

Comment:
Maybe we could extract this logic into a separate function since we're doing the same 8 times, even better if we use `sizeof(T)` rather than constants (`expectedLength`)

Reply:
Yep, let's do it. Still a bigger refactoring will be applied for SszLib later